contracts/contracts/RequestManager.sol

```diff
@@ -606,25 +630,25 @@ contract RequestManager is Ownable, LpWhitelist, RestrictedCalls, Pausable {
 
         if (claimValid) {
             // If claim is valid, all stakes go to the claimer
-            ethToTransfer = claimerStake + challengerStakeTotal;
             claimReceiver = claimer;
+            ethToTransfer = claimerStake + challengerStakeTotal;
         } else if (challengerStakeTotal > 0) {
-            // If claim is invalid, partial withdrawal by the sender
-            ethToTransfer = 2 * claim.challengersStakes[msg.sender];
-            claimReceiver = msg.sender;
+            // If claim is invalid, partial withdrawal by the claimResolver
+            claimReceiver = claimResolver;
+            ethToTransfer = 2 * claim.challengersStakes[claimReceiver];
```

Comment:
> Should we maybe consider requiring that the claimReceiver is actually one of the challengers?
claimReceiver could also be the claimer. Or did you actually mean `claimResolver` here?

> claimResolver -> participant
good idea! 

> claimReceiver -> stakeReceiver/stakeRecipient
I like that!

Reply:
> claimReceiver could also be the claimer. Or did you actually mean `claimResolver` here?

I meant challenger in this case specifically because we're accessing `challengersStakes`, but the same question could be asked for `claimResolver` -- do we need to ensure that the... `participant` really is either the claimer or a challenger?

Personally, I would add that restriction.

Reply:
> I meant challenger in this case specifically because we're accessing challengersStakes

we already do this exact check two lines below:

 `require(ethToTransfer > 0, "Challenger has nothing to withdraw");`

Reply:
But I would go for an additional check on a higher level, namely require that participant is actually a participant of this game.

Reply:
> we already do this exact check two lines below:
> 
> `require(ethToTransfer > 0, "Challenger has nothing to withdraw");`

Ah, good point, somehow I missed that. Felt like we should check for that earlier.